src/tasks/level10.ts

```diff
@@ -31,6 +31,26 @@ import { forceItemPossible, tryForceNC, tryPlayApriling } from "../engine/resour
 import { summonStrategy } from "./summons";
 import { args } from "../args";
 
+function batWingAirship(): boolean {
+  const turns = $location`The Penultimate Fantasy Airship`.turnsSpent;
```

Comment:
You should also check if bat wings are currently equipped

Reply:
I did a have check. I don't 100% understand how outfit is evaluated in relation to the choice settings, so since we try to equip them no matter what, having them implies they're equipped? I think.

Reply:
Assuming that it is forced elsewhere in the code, it's still not forced to be equipped in the directly relevant code.
In the future if kas does a rework, he's not going to know that the check that's isolated elsewhere in the code is related to this piece of code since there's no linkages.

That said, I don't believe it's 100% reliable as always equipped when this function runs, and this function will run even if you don't own the batwings iotm, which is a bad failstate.

Reply:
Oh, your latest commit can be heavily simplified.

```

function batWingAirship(): boolean {
  if (!have($item`bat wings`)) return false;
  
  // Rest of code as normal, no need to check about batwings

Reply:
Also, I'm not sure if this code is sane/stable because owning the iotm doesn't mean the iotm is equipped. Might be a better function for this, or might need to be in the choice adventure script.

Probably need kas to feedback on this, might even be the scenario where kas should move the choice adventure script into something where the choices can be resolved at choice runtime. Which is like greyday used to do, but could also be clumsy. Greyday had clumsy code for sure.

Reply:
Delayed choices (and delayed combat macros) always run after the outfit is actually equipped, specifically so that `haveEquipped` can be used here. So I think that would be more accurate and resilient

Reply:
So I changed this; apparently turnsSpent isn't always accurate, so choice adventure 6 *which should be available* isn't.

Unless you have some magic, we have a choice to make - Airship skip (saves 5 delay) or Saber in Airship.

Reply:
You could switch it to feel envy, especially if we can use bat wings for filthworms now